src/jit/lowerxarch.cpp

```diff
@@ -1110,6 +1116,23 @@ void Lowering::TreeNodeInfoInitCall(GenTreeCall* call)
         // computed into a register.
         if (!call->IsFastTailCall())
         {
+#ifdef _TARGET_X86_
+            // On x86, we need to generate a very specific pattern for indirect VSD calls:
+            //
+            //    3-byte nop
+            //    call dword ptr [eax]
+            //
+            // Where EAX is also used as an argument to the stub dispatch helper. Make
+            // sure that the call target address is computed into EAX in this case.
+            if (((call->gtFlags & GTF_CALL_VIRT_KIND_MASK) == GTF_CALL_VIRT_STUB) && (call->gtCallType == CT_INDIRECT))
+            {
+                assert(ctrlExpr->isIndir());
+
+                ctrlExpr->gtGetOp1()->gtLsraInfo.setDstCandidates(l, RBM_EAX);
```

Comment:
Thanks for the explanation.

So the set of source candidates for a particular node is the set of registers that are available at the point that the node is used, and the set of destination candidates for a node is the set of registers that the node may write? I would have expected that the set of source registers was the set of registers that the node may read--not the set that it may be read from. Now that it's been explained the terminology makes some sense, but it seems confusing given that `srcCount` refers to the number of registers read by a node.

> This makes me realize the same bug is present on amd64, because TreeNodeInfo::Initialize() is setting DstCandidates, in case of VSD calls.

Can you point me to the particular line?